client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -90,22 +90,41 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
                 return;
             }
 
-            if (!method.isTestMethod()) {
+            Class<?> cls = method.getTestMethod().getInstance().getClass();
+            final boolean isWebTestClass = cls.getAnnotation(WebTest.class) != null;
+            final boolean isMobileTestClass = cls.getAnnotation(MobileTest.class) != null;
+            final boolean isSeLionAnnotatedTestClass = isMobileTestClass || isWebTestClass;
+
+            // For non-session sharing, we only allow our annotation(s) on @Test methods.
+            // When this condition is true, we allow the session be created.
+            if (!method.isTestMethod() && !isSeLionAnnotatedTestClass) {
                 return;
             }
 
-            Class<?> cls = method.getTestMethod().getInstance().getClass();
-            boolean isWebTestClass = cls.getAnnotation(WebTest.class) != null;
-            boolean isMobileTestClass = cls.getAnnotation(MobileTest.class) != null;
-
-            if ((isWebTestClass || isMobileTestClass)) {
-                if (isLowPriority(method)) {
-                    // For session sharing tests, Need to create new session only for Test (Web or Mobile) with lowest
-                    // priority (first test) in the class.
-                    testSessionSharingRules(method);
-                } else {
+            // For session sharing, we only allow our annotation on the class.
+            // In this case the session can only be created in the @Test with the highest priority (first test, smallest
+            // number) or in a @BeforeClass
+            if (isSeLionAnnotatedTestClass) {
```

Comment:
The PR look good.

There is some opportunity for refactoring here.  Since all we need to have here is "isSeLionAnnotatedTestClass" we can put that computation in a private method